### DISCUSSION TITLE
Pause Timer and get the time remaining

### DISCUSSION THREAD

**Discussion by hernancasillas** [vydimitrov/react-countdown-circle-timer#114] [Q&A] [answered]
Hi, I'm trying to develop a flight booking app, when the user selects the seats and it's ready to fill their information, the timer starts, but I need to keep the current time and pass it to another screen so the new timer on the new screen for payment, starts with the time remaining that the previous timer has.

Or is there a way to pass the timer from screen to screen and not reseting every time?

I tried this to pause the timer, get the current time and save it into the state:
```javascript
<CountdownCircleTimer
                isPlaying={this.state.playing}
                duration={600}
                size={40}
                strokeWidth={2}
                renderAriaTime={(time) => {
                  if (this.state.getTime && !this.state.playing) {
                    this.setState({ time: time });
                  }
                }}
                colors={[
                  ["#004777", 0.4],
                  ["#F7B801", 0.4],
                  ["#A30000", 0.2],
                ]}
              >
                {children}
</CountdownCircleTimer>
<Button 
    onPress={()=>{
                      this.setState({
                        playing: false,
                        getTime: true,
                      });
                    //Then navigate to another screen to start the timer again with the value
    }}
/>
```
 but it says "Maximum update depth exceeded. This can happen when a component repeatedly calls setState inside componentWillUpdate or componentDidUpdate."

**Comment by vydimitrov** [comment, marked as answer]
Hey, I think you are overcomplicating it. Here is how I would do it:
- do not use props the way they are not supposed to be used. I refer to `renderAriaTime`
- do not pause the timer if you are going to unmount the component (it will get paused before unmounting)
- I would use `ref` rather than saving the remaining time in the state, like so:

```jsx
const duration = 600

class Timer {
  constructor() {
   // this is your ref to the remainigTime 
    this.remainigTime = duration
  }

  render() {
    return (
      <>
      <CountdownCircleTimer
            isPlaying
            duration={duration}
            initialRemainingTime={this.props.remainigTime} // this will be set to a value different than duration when on the next screen
            size={40}
            strokeWidth={2}
            colors={[
              ["#004777", 0.4],
              ["#F7B801", 0.4],
              ["#A30000", 0.2],
            ]}
      >
          {(timerProps) => {
             this.remainigTime = timerProps.remainigTime // here you save the remainigTime to the ref
            return children(timerProps)
          })}
     </CountdownCircleTimer>
     <Button 
       onPress={()=> this.props.onTimerStop(this.remainigTime))} // read remainigTime from ref and pass it when the button is clicked
     />
    </>
   )
 }
}
```

- this.props.remainigTime will be initially set  to the `duration`, for the next screen it will be the same as the `this.remainigTime`

Does this make sense? I am sorry if the class syntax is wrong. I don't remember the last time I had to write a class component.

**Reply by hernancasillas** [reply]
Yes, I managed to do it using a global variable. 
```javascript
{(timerProps) => {
    global.time= timerProps.remainingTime; // here you save the remainigTime to the ref
    return children(timerProps);
}}
```

Thank you so much!
